Microsoft.HealthVault.Fhir/Vocabularies/VocabToHealthVaultHelper.cs

```diff
@@ -0,0 +1,90 @@
+
+using System;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Vocabularies
+{
+    internal static class VocabToHealthVaultHelper
+    {        
+        internal static Type DetectHealthVaultTypeFromObservation(Observation observation)
+        {
+            if (observation.Code != null && observation.Code.Coding != null)
+            {
+                foreach (var code in observation.Code.Coding)
+                {
+                    switch (code.System.ToLowerInvariant())
+                    {
+                        case VocabularyUris.SnomedCd:
+                            return DetectFromSnomedCd(code.CodeElement.Value);
+                        case VocabularyUris.Loinc:
+                            return DetectFromLoincCodes(code.CodeElement.Value);
+                        case VocabularyUris.HealthVaultVocabulariesUri:
+                            return DetectFromHealthVaultCode(code.CodeElement.Value);
+                    }
+                }
+            }
+
+            throw new NotSupportedException();
+        }
+
+        private static Type DetectFromSnomedCd(string code)
+        {
+            switch (code.ToLowerInvariant())
+            {
+                case "434912009":
+                    return typeof(BloodGlucose);
+            }
+
+            throw new NotSupportedException("The provided code is not supported");
+        }
+
+        private static Type DetectFromHealthVaultCode(string code)
+        {
+            var vocab = code.Split(':');
+
+            if (vocab.Length == 2)
+            {
+                var vocabName = vocab[0];
+                var vocabValue = vocab[1];
+
+                switch (vocabName.ToLowerInvariant())
+                {
+                    case HealthVaultVocabularies.VitalStatistics:
+                        switch (vocabName.ToLowerInvariant())
+                        {
+                            case "wgt":
+                                return typeof(Weight);
+                        }
+                        break;
+                    case HealthVaultVocabularies.BloodGlucoseMeasurementContext:
+                    case HealthVaultVocabularies.BloodGlucoseMeasurementType:
+                        return typeof(BloodGlucose);
+                }
+
+            }
+
+            throw new NotSupportedException("The provided code is not supported");
+        }
+        private static Type DetectFromLoincCodes(string code)
```

Comment:
This is not done yet. As part of the pending elements for next sprint is to improve on how we map this. We would like to use a file driven mechanism

---
In reply to: [124311254](https://github.com/Microsoft/healthvault-fhir-library/pull/3#discussion_r124311254) [](ancestors = 124311254)